ep/build-velox/src/build_velox.sh

```diff
@@ -128,7 +128,8 @@ function compile {
     COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_BENCHMARKS=ON"
   fi
   if [ $ENABLE_GPU == "ON" ]; then
-    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_GPU=ON -DVELOX_ENABLE_CUDF=ON"
+    # the cuda default options are for Centos9 image from Meta
+    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_GPU=ON -DVELOX_ENABLE_CUDF=ON -DCMAKE_CUDA_ARCHITECTURES=70 -DCMAKE_CUDA_COMPILER=/usr/local/cuda-12.8/bin/nvcc"
```

Comment:
I use the ENV here but without CUDA_COMPILER https://github.com/apache/incubator-gluten/pull/9333/files#diff-395a263b6f69bd7ef4991face3666e100acdd60e47f4b28e0737adbbf5fb945aR12,is all the environment CUDA COMPILER this path? Do we need to make it an environment variable in docker image?

Reply:
In my tests it will report `cuda identifier is undefined in device code` if not setting these two variables. Note the GHA runner does not have NV GPU installed.

`CUDA_ARCHITECTURES`is able to read from `CMAKE_CUDA_ARCHITECTURES`
https://cmake.org/cmake/help/v3.28/prop_tgt/CUDA_ARCHITECTURES.html
however it seems there's no similar variable like `CUDA_COMPILERS`